### PR TITLE
 New Ftr: allow user set custom params to register to registry

### PR DESCRIPTION
**Pull request by hxmhlt**
**What this PR does**:

 New Ftr: allow user set custom params to register to registry
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
User can set references & services params in consumer.yml & provider.yml
```

### CODE REVIEW

**Comment by wongoo** [comment]
LGTM

File: remoting/zookeeper/listener.go [hunk 1]
```diff
@@ -129,14 +129,14 @@ func (l *ZkEventListener) handleZkNodeEvent(zkPath string, children []string, li
 			continue
 		}
 		// listen l service node
-		go func(node string) {
+		go func(node, childNode string) {
```
Comment (gaoxinge) [inline remoting/zookeeper/listener.go]: `zkPath` may also add to parameter?

File: remoting/zookeeper/listener_test.go [hunk 2]
```diff
@@ -86,32 +87,36 @@ func TestListener(t *testing.T) {
...
 	client      *ZookeeperClient
 	changedData string
+	wait        *sync.WaitGroup
 }
 
 func (m *mockDataListener) DataChange(eventType remoting.Event) bool {
 	fmt.Println(eventType)
 	m.eventList = append(m.eventList, eventType)
 	if eventType.Content == m.changedData {
-		m.client.Wait.Done()
```
Comment (gaoxinge) [inline remoting/zookeeper/listener_test.go]: Why not just change `m.client.Wati.Done` to `m.client.Close`?

Comment (hxmhlt) [reply remoting/zookeeper/listener_test.go]: If I want to test another data change event, close client may not be a good idea. Instead , I define a wait group for one data change event

File: remoting/zookeeper/listener.go (hunk 1 above)
Comment (hxmhlt) [reply remoting/zookeeper/listener.go]: @AlexStocks why add childNode ,just use in log?

File: remoting/zookeeper/listener_test.go (hunk 2 above)
Comment (gaoxinge) [reply remoting/zookeeper/listener_test.go]: @hxmhlt Ok. I think your solution may be more suitable.

**Pull request merged**